### PR TITLE
add color function for common.Player

### PR DESCRIPTION
**Pull request by Julien2313**
(#248)

Worked perfectly on a demo I tested =>

> CT Black Fart Yellow
> CT KMA Purple
> T Ryba Orange
> CT Xelf Blue
> CT Cheuveulu Orange
> T Nozyyy🦁 Yellow
> T Slink1019 Blue
> T Kamik Green
> T Stog CS.DEALS Purple
> CT socio* linux <3 Green

Let me know how should I test that ?

### CODE REVIEW

File: pkg/demoinfocs/common/common.go [hunk 1]
```diff
@@ -239,3 +239,26 @@ func ConvertSteamID32To64(steamID32 uint32) uint64 {
...
+const (
+	Grey Color = iota - 1
+	Yellow
+	Purple
+	Green
+	Blue
+	Orange
+)
+
+var strColors = map[int]string{-1: "Grey", 0: "Yellow", 1: "Purple", 2: "Green", 3: "Blue", 4: "Orange"}
```
Comment (markus-wa) [inline pkg/demoinfocs/common/common.go:256]:
```suggestion
var strColors = map[Color]string{
	Grey: "Grey",
	Yellow: "Yellow",
	Purple: "Purple",
	Green: "Green",
	Blue: "Blue",
	Orange: "Orange",
}
```

File: pkg/demoinfocs/common/common.go [hunk 2]
```diff
@@ -239,3 +239,26 @@ func ConvertSteamID32To64(steamID32 uint32) uint64 {
...
+
+func (c Color) String() string {
+	if _, exists := strColors[int(c)]; !exists {
+		return "unknownColor"
```
[the remark this answers is not in the document]
Comment (markus-wa) [reply pkg/demoinfocs/common/common.go:260]:
```suggestion
		return "Unknown-Color"
```

**Review by markus-wa — COMMENTED** [review]
thanks for the PR.

> Let me know how should I test that ?

I think a unit test should do it, see https://github.com/markus-wa/demoinfocs-golang/blob/master/pkg/demoinfocs/common/player_test.go#L356-L360 :slightly_smiling_face: 

I'll have to take a look at the automated codereviews (golangci-lint and reviewdog) since they aren't catching some stuff that they really should (e.g. gomnd isn't reporting). hope this won't delay this too much.

**Comment by Julien2313** [comment]
I'll check to do the tests later today ;)

No problem for the delay.

**Comment by Julien2313** [comment]
image

Finally here ! ha ha

**Comment by Julien2313** [comment]
`VectorVal, ArrayVal, StringVal, FloatVal are missing in PropertyValue (exhaustivestruct)` Mmmh, is that normal ?

**Comment by markus-wa** [comment]
@Julien2313 - I think I disabled that check on master, just merged that back in. Let's see what the new build says 🤞

**Comment by Julien2313** [comment]
RIP ha ha

**Comment by markus-wa** [comment]
oops that was just on a branch (#249)

let's try that again

**Comment by markus-wa** [comment]
merged. I've also put out a new release [v2.4.0](https://github.com/markus-wa/demoinfocs-golang/releases/tag/v2.4.0) with this and a few other changes

**Comment by Julien2313** [comment]
Nice ! Thanks

**Pull request merged**